src/components/vm/disks/diskAdd.jsx

```diff
@@ -365,511 +359,499 @@ const CustomPath = ({ idPrefix, onValueChanged, device, validationFailed, hideDe
     );
 };
 
-export class AddDiskModalBody extends React.Component {
-    static contextType = DialogsContext;
-
-    constructor(props) {
-        super(props);
-        this.state = {
-            validate: false,
-            dialogLoading: true,
-            customDiskVerificationFailed: false,
-            customDiskVerificationMessage: null,
-            verificationInProgress: false,
-            file: null,
-        };
-        this.onValueChanged = this.onValueChanged.bind(this);
-        this.dialogErrorSet = this.dialogErrorSet.bind(this);
-        this.onAddClicked = this.onAddClicked.bind(this);
-        this.onInsertClicked = this.onInsertClicked.bind(this);
-        this.existingVolumeNameDelta = this.existingVolumeNameDelta.bind(this);
-        this.validateParams = this.validateParams.bind(this);
-    }
+const sortFunction = (poolA, poolB) => poolA.name.localeCompare(poolB.name);
+
+export const AddDiskModalBody = ({ disk, idPrefix, isMediaInsertion, vm, vms, supportedDiskBusTypes }) => {
+    const [customDiskVerificationFailed, setCustomDiskVerificationFailed] = useState(false);
+    const [customDiskVerificationMessage, setCustomDiskVerificationMessage] = useState(null);
+    const [dialogError, setDialogError] = useState(null);
+    const [dialogErrorDetail, setDialogErrorDetail] = useState(null);
+    const [diskParams, setDiskParams] = useState({
+        cacheMode: 'default',
+        device: "disk",
+        file: "",
+        format: "",
+        serial: "",
+        size: 1,
+        unit: units.GiB.name,
+        volumeName: "",
+    });
+    const [mode, setMode] = useState(isMediaInsertion ? CUSTOM_PATH : CREATE_NEW);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1303-20231117-090529-31f818c6-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)